nomad/client_stats_endpoint.go

```diff
@@ -74,12 +74,17 @@ func (s *ClientStats) Stats(args *structs.ClientStatsRequest, reply *structs.Cli
 			return fmt.Errorf("Unknown node %q", args.NodeID)
 		}
 
-		// TODO Handle forwarding to other servers
-		return ErrNoNodeConn
+		// Determine the Server that has a connection to the node.
+		srv, err := s.srv.serverWithNodeConn(args.NodeID)
+		if err != nil {
+			return err
+		}
+
```

Comment:
should also error when srv is nil here

Reply:
I added it here since I can put a better error. So good catch